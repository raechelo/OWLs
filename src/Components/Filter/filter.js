import React, { Component } from 'react';

class Filter extends Component {
  constructor() {
    super();
    this.state = {
      category: ''
    }
  }

  updateCategory = () => {
    this.setState( {
      category: this.props.category,
      questions: this.props.studySet
    } )
  }

  filterBy = (e) => {
    let cat = e.target.textContent
    let filteredQs = this.props.studySet.filter(q => q.category === cat);
    this.setState( {
      category: cat,
      questions: filteredQs
    } )
  }

  render () {

    return (
      <div>
        <button onClick={this.filterBy}>React</button>
        <button onClick={this.filterBy}>Redux</button>
        <button onClick={this.filterBy}>Router</button>
      </div>
    )
  }
}

export default Filter;