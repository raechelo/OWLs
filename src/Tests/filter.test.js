import React from 'react';
import Filter from '../Components/Filter/filter';
import { shallow } from 'enzyme';

describe('Filter', () => {

  const mockData = [
    {
      "question": "In which lifecycle method should you fetch APIs?",
      "correctAnswer": "componentDidMount",
      "allAnswers": ["componentDidMount", "componentWillMount", "componentDoesMount", "componentDidUnmount"],
      "category": "React",
      "id": 1
    },
    {
      "question": "What would be the proper way to add new values to an array stored in state?",
      "correctAnswer": "this.setState({myArray: newValues})",
      "allAnswers": ["this.setState({myArray: newValues})", "this.setState(newValues)", "this.state.myArray.push(newValues)", "myArray.push(newValues)"],
      "category": "React",
      "id": 2
    }
  ]

  let wrapper;

  beforeEach(() => {

    wrapper = shallow(
      <Filter studySet={mockData} />
    )
  });

  it('should match the snapshot with all the data passed in', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should have a proper default state', () => {
    expect(wrapper.state()).toEqual({ category: '' } )
  });

  it('should filter categories', () => {
    expect(wrapper.state('category')).toEqual( '' );
    wrapper.instance().filterBy( {target: {textContent: 'React' } } );
    expect(wrapper.state('category')).toEqual( 'React' );
  });
  
})

