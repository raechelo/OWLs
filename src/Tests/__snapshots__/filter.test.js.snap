// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Filter should match the snapshot with all the data passed in 1`] = `
<div>
  <button
    onClick={[Function]}
  >
    React
  </button>
  <button
    onClick={[Function]}
  >
    Redux
  </button>
  <button
    onClick={[Function]}
  >
    Router
  </button>
</div>
`;
